tests/e2e/performance/test_pvc_snapshot_performance.py

```diff
@@ -0,0 +1,275 @@
+    """
+
+    @pytest.fixture()
+    def base_setup(
+        self, request, interface_iterate, storageclass_factory,
+        pvc_factory, pod_factory, pvc_size
+    ):
+        """
+        A setup phase for the test - creating resources
+
+        Args:
+            interface_iterate: A fixture to iterate over ceph interfaces
+            storageclass_factory: A fixture to create everything needed for a
+                storageclass
+            pvc_factory: A fixture to create new pvc
+            pod_factory: A fixture to create new pod
+            pvc_size: The size of the PVC in Gi
+
+        """
+        self.interface = interface_iterate
+        self.sc_obj = storageclass_factory(self.interface)
+
+        self.pvc_obj = pvc_factory(
+            interface=self.interface,
+            size=pvc_size,
+            status=constants.STATUS_BOUND
+        )
+
+        self.pod_obj = pod_factory(
+            interface=self.interface,
+            pvc=self.pvc_obj,
+            status=constants.STATUS_RUNNING
+        )
+
+    @pytest.mark.parametrize(
+        argnames=["pvc_size"],
+        argvalues=[
+            pytest.param(*['1']),
+            pytest.param(*['10']),
+            pytest.param(*['100'])
+        ]
+    )
+    @pytest.mark.usefixtures(base_setup.__name__)
+    def test_pvc_snapshot_performance(self, teardown_factory, pvc_size):
+        """
+        1. Run I/O on a pod file.
+        2. Calculate md5sum of the file.
+        3. Take a snapshot of the PVC and measure the time of creation.
+        4. Restore From the snapshot and measure the time
+        5. Attach a new pod to it.
+        6. Verify that the file is present on the new pod also.
+        7. Verify that the md5sum of the file on the new pod matches
+           with the md5sum of the file on the original pod.
+
+        This scenario run 3 times and report all results
+        Args:
+            teardown_factory: A fixture to destroy objects
+            pvc_size: the size of the PVC to be tested - parametrize
+
+        """
+
+        tests_numbers = 3  # number of tests to run
+
+        # Getting the total Storage capacity
+        ceph_cluster = CephCluster()
+        ceph_capacity = ceph_cluster.get_ceph_capacity()
+
+        log.info(f'Total capacity size is : {ceph_capacity}')
+        log.info(f'PVC Size is : {pvc_size}')
+        log.info(f'Needed capacity is {int(int(pvc_size) * 5)}')
+        if int(ceph_capacity) < int(pvc_size) * 5:
+            log.error(
+                f'PVC size is {pvc_size}GiB and it is too large for this system'
+                f' which have only {ceph_capacity}GiB'
+            )
+            return
+        # Calculating the file size as 25% of the PVC size
+        # in the end the PVC will be 75% full
+        filesize = self.pvc_obj.size * 0.25
+        # Change the file size to MB and from int to str
+        file_size = f'{int(filesize * 1024)}M'
```

Comment:
@Avilir  Will the snapshot and restore time also be affected by the number of files being present in the PVC? If so, we might need to add another test to measure performance based on the number of files in the PVC. WDYT?

Reply:
AFIK, it only depend on the total data on the PVC regardless the number of file.

Reply:
For cephFS, I believe there is going to be an overhead. Let's check with the cephFS engineering team. But in any case, that would be a new test, and let's not hold this PR.

Reply:
CephFS clone is a deep copy of files and directories from the source volume/snapshot. Thus, as the file count or directory count increases the time to clone would also increase.

RBD clones are cheaper in that, the clone only refers to the parent snapshot/image.

CephFS and RBD snapshots are not dependent on the size of the image or content of the subvolume, these are CoW/RoW based snapshots and hence are a constant set of operations that need to be performed for an image or subvolume irrespective of the content within.